src/schema/application/configuration/schema-configuration.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+
+// phpcs:disable Yoast.NamingConventions.NamespaceName.TooLong
+namespace Yoast\WP\SEO\Schema\Application\Configuration;
+
+use Easy_Digital_Downloads;
+use SeriouslySimplePodcasting\Integrations\Yoast\Schema\PodcastEpisode;
+use TEC\Events\Integrations\Plugins\WordPress_SEO\Events_Schema;
+use WP_Recipe_Maker;
+use WPSEO_Addon_Manager;
+use Yoast\WP\SEO\Helpers\Product_Helper;
+use Yoast\WP\SEO\Helpers\Woocommerce_Helper;
+
+/**
+ * Responsible for the schema configuration.
+ */
+class Schema_Configuration {
+
+	/**
+	 * The WooCommerce helper.
+	 *
+	 * @var Woocommerce_Helper
+	 */
+	private $woocommerce_helper;
+
+	/**
+	 * The product helper.
+	 *
+	 * @var Product_Helper
+	 */
+	private $product_helper;
+
+	/**
+	 * Schema_Configuration constructor.
+	 *
+	 * @param Woocommerce_Helper $woocommerce_helper The WooCommerce helper.
+	 * @param Product_Helper     $product_helper     The product helper.
+	 */
+	public function __construct( Woocommerce_Helper $woocommerce_helper, Product_Helper $product_helper ) {
+		$this->woocommerce_helper = $woocommerce_helper;
+		$this->product_helper     = $product_helper;
+	}
+
+	/**
+	 * Returns the schema configuration.
+	 *
+	 * @return array<string, bool|array<string, array<string, bool|string>>>
+	 */
+	public function get_configuration(): array {
+		return [
+			'isSchemaDisabledProgrammatically' => $this->is_schema_disabled_programmatically(),
+			'schemaApiIntegrations'            => $this->get_schema_api_integrations(),
+		];
+	}
+
+	/**
+	 * Gets the schema API integrations status.
+	 *
+	 * @return array<string, array<string, bool|string>> The schema API integrations status.
+	 */
+	private function get_schema_api_integrations(): array {
+		$addon_manager = new WPSEO_Addon_Manager();
```

Comment:
We shouldn't instantiate class like that, but rather make it part of the DI instead. For example, check out the usage of the addon manager class in eg. the `Indexing_Error_Presenter` class.